proposals/semi-auto-properties.md

```diff
@@ -108,6 +108,24 @@ Open LDM questions:
 
 1. If a type does have an existing accessible `field` symbol in scope (like a field called `field`) should there be any way for an auto-prop to still use `field` internally to both create and refer to an auto-prop field.  Under the current rules there is no way to do that.  This is certainly unfortunate for those users, however this is ideally not a significant enough issue to warrant extra dispensation.  The user, after all, can always still write out their properties like they do today, they just lose out from the convenience here in that small case.
 
+2. Should initializers use the backing field or the property setter? If the latter, what about `public int P { get => field; } = 5;`?
+
+3. Can the field be written into in a property without setter?
```

Comment:
I definitely hope so: `public string Foo => field ??= GetFoo();`

Reply:
Yes. This is a core scenario. It has to work.

Reply:
For '2' I think it should use the setter.

Reply:
Should I remove `3` from the PR if the answer is clear and doesn't need to go in LDM?

> For '2' I think it should use the setter.

And if there is no setter? Should be error or use the field directly?

Reply:
Thus far, constructor property assignments fall back to using the field directly if there is no setter.

I'm feeling skeptical that property initializers calling the setter will be a nice surprise.

Reply:
> Should I remove 3 from the PR if the answer is clear and doesn't need to go in LDM?

Yup

Reply:
> And if there is no setter? Should be error or use the field directly?

The latter.  Just like a normal auto prop without a setter.

Reply:
> And if there is no setter? Should be error or use the field directly?

The latter.  Just like a normal auto prop without a setter.

Reply:
> A principle in the proposal is that get; and get => field; have identical semantics.

Yup yup

Reply:
With `public int Foo { get; set => DoStuff(ref field, value); } = 1;`, I think we'll want a way to initialize the field without executing `DoStuff`, without losing the ability to use an autoprop/`field`.

There's already a way to go the other way if you *do* want to explicitly execute the setter: turn the property initializer into a constructor assignment.

Reply:
For `2`, I agree with @jnm2 that it's more flexible if the initializer uses the backing field.

For `3`, I removed it from this PR and opened a spec change: https://github.com/dotnet/csharplang/pull/5575

Reply:
I'm coming around to this position.  Though i think it's def got a downside (as well as a virtue) that field-assign and constructor-assign mean separate things.

For now, i think we shoudl make it so that:

1. property-init means field-assign (it legit can't mean 'call setter' unless we do some interesting emit stuff).
2. constructor-assign means 'call setter'.
3. we bring this to LDM to verify this is the behavior we want.

Reply:
Additionally, if the property initializer does mean calling the setter, then people who actually can't invoke the setter are pushed over the cliff and have to refactor to declaring their own field. This pain could be mitigated by implementing property-scoped fields, making this syntax possible:

```cs
public int Foo { int field = 1; get; set => Set(ref field, value); }
```

Even with that ability, I would personally still prefer this syntax to not call the setter:
```cs
public int Foo { get; set => Set(ref field, value); } = 1;
```

Reply:
Adding this info in https://github.com/dotnet/csharplang/pull/5837